src/Compilers/Core/Portable/MetadataReader/MetadataDecoder.cs

```diff
@@ -1095,15 +1109,26 @@ private void DecodeParameterOrThrow(ref BlobReader signatureReader, /*out*/ ref
         {
             bool refersToNoPiaLocalType;
 
+            var isConstFound = false;
             SignatureTypeCode typeCode;
-            info.CustomModifiers = DecodeModifiersOrThrow(ref signatureReader, out typeCode);
+            info.CustomModifiers = DecodeModifiersOrThrow(ref signatureReader, out typeCode, acceptRequiredModifier: type =>
+            {
+                var isConst = IsConstModifierType(type);
+                isConstFound |= isConst;
+                return isConst;
+            });
 
             if (typeCode == SignatureTypeCode.ByReference)
             {
                 info.IsByRef = true;
-                info.RefCustomModifiers = info.CustomModifiers; 
+                info.RefCustomModifiers = info.CustomModifiers;
                 info.CustomModifiers = DecodeModifiersOrThrow(ref signatureReader, out typeCode);
             }
+            else if (isConstFound)
+            {
+                // This cannot be placed on CustomModifiers, just RefCustomModifiers
+                throw new UnsupportedSignatureContent();
+            }
```

Comment:
Is `isConstFound` needed or could the check above be replaced by `else if (info.CustomModifiers.Length > 0)`? Then the delegate to `DecodeModifiersOrThrow` can be simplified. Similar comment for `isVolatileFound` in `DecodeFieldSignature`.